src/locale/localed.c

```diff
@@ -283,7 +284,7 @@ static int process_locale_list_item(
 
                 if (!locale_is_valid(e))
                         return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Locale %s is not valid, refusing.", e);
-                if (locale_is_installed(e) <= 0)
+                if ((locale_is_installed(e) <= 0) && !use_localegen)
```

Comment:
hmm, locale_is_installed() is not free and kinda pointless to call if use_localegen is true. Hence, please swap the two sides of the `&&`, so that we only call it if use_locale_gen is false